### ISSUE TITLE
Colour picker only picks only color from the desktop background on MacOS

### ISSUE THREAD

**Issue by Jirodica** [ThePacielloGroup/CCAe#163]
**Describe the bug**
When you make use of the colour picker the focus is on desktop background and it is not possible to pick a colour from the image you want to test.

**Desktop (please complete the following information):**
 - macOS Catalina version 10.15.4

**To Reproduce**
choose the colour picker for the foreground colour and move the picker to the image fror picking the right colour. 

**Expected behavior**

I should pickup a colour from the choosen image and not from the desktop background

**Screenshots**
![IMG_3372](https://user-images.githubusercontent.com/9657804/78263120-d78d4c00-7501-11ea-8565-8c9e18c3e80b.jpg)

**Additional context**

**Comment by matatk** [comment]
I had the same problem, but noticed that after I quit CCA, there was a macOS dialog behind the CCA window that was asking for permission to access the 'screen recording' API.

After I granted permission (System Preferences > Security & Privacy > Privacy > Screen Recording) and restarted CCA, it worked.

**Comment by matatk** [comment]
I've just tried it on another machine and got the same results—granting permissions worked, but the dialog had appeared _behind_ the CCA window. It seems the dialog appears when the user first activates the colour picker tool.

![permissions dialog](https://user-images.githubusercontent.com/1125522/80464035-18c22180-8931-11ea-89d4-5a9a06fbf2b5.png)

Therefore I think two things need addressing:

1. Making sure the requester pops up on top.

2. Ideally making it clearer as to _why_ this permission is needed. Perhaps the text in the dialog can be changed?

**Comment by ferllings** [comment]
I'm not sure if this is feasible: This is a MacOS security dialog. We don't have any control over it.
And It is only happening on latest MacOS version.
It's not displayed on front probably because CCA is already on "display on top" mode.

>     1. Making sure the requester pops up on top. 
>     2. Ideally making it clearer as to _why_ this permission is needed. Perhaps the text in the dialog can be changed?

**Comment by matatk** [comment]
@ferllings yeah, I was wondering about that. Was hoping there may be APIs to either request, or at least check for, that permission. As you say though, it all seems very new.

I did a quick search on StackOverflow and found a few questions, though most seem to be Objective-C-related. However, I found [one question that mentions Electron](https://stackoverflow.com/q/59529585/1485308). Someone recommended an npm package that has some Swift code in it. I think it's working by trying to grab a pixel and seeing if it got `nil` (meaning permssions were not granted).

Not sure if you'd want to look at that, given the extra complications to the code, and how new this all is. I wonder if, on macOS, it is worth putting up some UI when CCA starts to manage users' expectations. If "display on top" mode is the default, it may be worth it, though given that you mentioned it, I'm thinking maybe I turned it on ages ago and have forgot about it :-).

**Comment by Jerred13** [comment]
Hello,  I am having this problem also, but this pop up doesnt appear to show anywhere.  I have tried restarting, tried uninstall and re-install.  Any ideas?

**Comment by ferllings** [comment]
Please try this: https://github.com/ThePacielloGroup/CCAe/issues/163#issuecomment-619379550

**Comment by igotabeme2** [comment]
> Please try this: [#163 (comment)](https://github.com/ThePacielloGroup/CCAe/issues/163#issuecomment-619379550)

Thank you! This solution did work for me, however I did not have a dialog window regarding the issue. I had to manually go into the system preferences and set the 'Screen recording api'. Once I turned that on and restarted CCA, it is working as expected.

Thank you all!

**Comment by igotabeme2** [comment]
> Thank you! This solution did work for me, however I did not have a dialog window regarding the issue. I had to manually go into the system preferences and set the 'Screen recording api'. Once I turned that on and restarted CCA, it is working as expected.
> 
> Thank you all!



> I had the same problem, but noticed that after I quit CCA, there was a macOS dialog behind the CCA window that was asking for permission to access the 'screen recording' API.
> 
> After I granted permission (System Preferences > Security & Privacy > Privacy > Screen Recording) and restarted CCA, it worked.

**Comment by ferllings** [comment]
Fixed with use of Eyedropper API

**Comment by pmtokle** [comment]
> I had the same problem, but noticed that after I quit CCA, there was a macOS dialog behind the CCA window that was asking for permission to access the 'screen recording' API.
> 
> After I granted permission (System Preferences > Security & Privacy > Privacy > Screen Recording) and restarted CCA, it worked.

Thanks a lot!
